chips/sam4l/src/ast.rs

```diff
@@ -246,15 +263,17 @@ impl<'a> Alarm for Ast<'a> {
         unsafe { (*self.regs).cv.get() }
     }
 
-    fn set_alarm(&self, tics: u32) {
-        self.disable();
+    fn set_alarm(&self, mut tics: u32) {
         while self.busy() {}
         unsafe {
+            let now = (*self.regs).cv.get();
```

Comment:
I thought one of Rust's "things" was that the `.` operator also dereferenced pointers when necessary. Is there a reason why you are manually dereferencing it instead? (if so, please comment the line!)